src/ferc_eia_match/name_cleaner.py

```diff
@@ -0,0 +1,329 @@
+"""This module contains the implementation of CompanyNameCleaner class from OS-Climate's financial-entity-cleaner package."""
+
+import enum
+import json
+import os
+import re
+from typing import Any
+
+import pandas as pd
+
+CLEANING_RULES_DICT = {
+    "remove_email": [" ", r"\S*@\S*\s?"],
+    "remove_url": [" ", r"https*\S+"],
+    "remove_word_the_from_the_end": [" ", r"the$"],
+    "place_word_the_at_the_beginning": [" ", r"the$"],
+    "remove_www_address": [" ", r"https?://[.\w]{3,}|www.[.\w]{3,}"],
+    "enforce_single_space_between_words": [" ", r"\s+"],
+    "replace_amperstand_by_AND": [" and ", r"&"],
+    "add_space_between_amperstand": [" & ", r"&"],
+    "replace_amperstand_between_space_by_AND": [" and ", r"\s+&\s+"],
+    "replace_hyphen_by_space": [" ", r"-"],
+    "replace_hyphen_between_spaces_by_single_space": [" ", r"\s+-\s+"],
+    "replace_underscore_by_space": [" ", r"_"],
+    "replace_underscore_between_spaces_by_single_space": [" ", r"\s+_\s+"],
+    "remove_all_punctuation": [" ", r"([^\w\s])"],
+    "remove_punctuation_except_dot": [" ", r"([^\w\s.])"],
+    "remove_mentions": [" ", r"@\S+"],
+    "remove_hashtags": [" ", r"#\S+"],
+    "remove_numbers": [" ", r"\w*\d+\w*"],
+    "remove_text_puctuation": [" ", r'\;|\:|\,|\.|\?|\!|"'],
+    "remove_text_puctuation_except_dot": [" ", r'\;|\:|\,|\?|\!|"'],
+    "remove_math_symbols": [" ", r"\+|\-|\*|\>|\<|\=|\%"],
+    "remove_math_symbols_except_dash": [" ", r"\+|\*|\>|\<|\=|\%"],
+    "remove_parentheses": ["", r"\(|\)"],
+    "remove_brackets": ["", r"\[|\]"],
+    "remove_curly_brackets": ["", r"\{|\}"],
+    "remove_single_quote_next_character": [" ", r"'\w+"],
+    "remove_single_quote": [" ", r"'"],
+    "remove_double_quote": [" ", r'"'],
+    "remove_words_in_parentheses": [" ", r"\([^()]*\)"],
+    "repeat_remove_words_in_parentheses": [" ", r"remove_words_in_parentheses"],
+}
+
+DEFAULT_COMPANY_CLEANING_RULES = [
+    "replace_amperstand_between_space_by_AND",
+    "replace_hyphen_between_spaces_by_single_space",
+    "replace_underscore_by_space",
+    "replace_underscore_between_spaces_by_single_space",
+    "remove_text_puctuation_except_dot",
+    "remove_math_symbols",
+    "remove_words_in_parentheses",
+    "remove_parentheses",
+    "remove_brackets",
+    "remove_curly_brackets",
+    "enforce_single_space_between_words",
+]
+
+
+def load_json_file(file_to_read: str) -> Any:
```

Comment:
If this function is primarily going to be used to read the JSON file(s) that are part of the package, it should probably be using `importlib.resources` to access that data.  If it's also going to be consuming user-provided JSON, then there might need to be some awareness of both methods of file access.

Reply:
Moved this JSON file to `package_data` and switched to using `importlib.resources`.